src/components/PopupWithForm.js

```diff
@@ -2,29 +2,51 @@ import Popup from "./Popup";
 
 export default class PopupWithForm extends Popup {
   constructor({ popupSelector, handleFormSubmit,}) {
-    super({ popupSelector });
-    this._popupForm = this._popupElement.querySelector(".modal__form");
+    super({ popupSelector});
     this._handleFormSubmit = handleFormSubmit;
+    this._popupForm = this._popupElement.querySelector('.modal__form');
+    this._inputList = this._popupElement.querySelectorAll('.modal__input');
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonContent = this._submitButton.textContent;
+
   }
 
   _getInputValues() {
-    this._inputList = this._popupElement.querySelectorAll(".modal__input");
     this._formValues = {};
-    this._inputList.forEach((input) => {
-      this._formValues[input.name] = input.value;
-    });
+    this._inputList.forEach(input => this._formValues[input.name] = input.value);
 
     return this._formValues;
   }
 
+  _submit(evt) { //this method is to be called when form is submited
+    evt.preventDefault();
+    console.log(this);
+    this._handleFormSubmit(this._getInputValues()); //call external callback _handleFormSubmit
+    this.close();
```

Comment:
` this.close();`


Please, delete it


You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

Reply:
* Another remark hasn't been corrected https://github.com/UffyLane/se_project_aroundtheus/pull/1#discussion_r2060649276